### PR TITLE
Update pull request template with more details

### PR DESCRIPTION
**Pull request by captainsafia**
Updating our pull request information to have more helpful info for new contributors per a recent discussion.

Also clarifying that "Summary of changes" should be in the PR title so it shows up as the commit message when we squash and merge.

Thoughts on other things we should include in the checklist?

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
+
+<!-- Thank you for submitting a pull request to our repo. -->
+
+<!-- If this is your first PR in the ASP.NET Core repo, please run through the checklist 
+below to ensure a smooth review and merge process for your PR. -->
+
+- [ ] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/master/CONTRIBUTING.md).
+- [ ] You've included unit or integration tests for your change, where applicable.
+- [ ] You've included inline docs for your change, where applicable.
+- [ ] There's an open issue for the PR that you are making. If you'd like to purpose a new feature or change, please open an issue to discuss the change or find an existing issue.
```
Comment (BrennanConroy) [inline .github/pull_request_template.md:10]:
```suggestion
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
```

**Review by BrennanConroy — APPROVED** [review]

**Review by Pilchie — APPROVED** [review]

**Review by dougbu — APPROVED** [review]

**Review by javiercn — APPROVED** [review]

**Comment by captainsafia** [comment]
@dotnet/aspnet-build Can I get your help merging this? Looks like the CI checks are in a weird state because this PR doesn't change any code.

**Comment by dougbu** [comment]
I need to revert my change to ignore doc-only changes☹️

**Pull request merged**